mechanicalsoup/stateful_browser.py

```diff
@@ -152,6 +152,24 @@ def open_relative(self, url, *args, **kwargs):
         """
         return self.open(self.absolute_url(url), *args, **kwargs)
 
+    def refresh(self):
+        """Reloads the current page with the same request as originally done.
+        Works only if page was loaded using ``StatefulBrowser`` methods.
```

Comment:
I'd omit this "works only", since most methods of StatefulBrowser already behave like this.

Reply:
I think he was referring to the fact that it won't reload the page from a `StatefulBrowser.get`, even though called on a `StatefulBrowser` object, since it's using a `Browser` (stateless) method. I'm not sure how to best phrase this.

I almost want to get rid of the `Browser` class and have everything be "stateful" to avoid this kind of ambiguity, but... that's a much longer discussion for a much later time. ;)

Reply:
@hemberger is right about my intention. Until I started to created the PR I did not know that there was a different behavior when opening pages via `StatefulBrowser` or `Browser`. Maybe I just over-read this somewhere.

And yes, I would also support to get rid of `Browser` :D

Reply:
> I think he was referring to the fact that it won't reload the page from a StatefulBrowser.get

Yes, but so does following a relative link or submitting a form with a relative action. It's not specific to `reload`.

> I almost want to get rid of the Browser class and have everything be "stateful" to avoid this

Let's start a separate discussion: #190

Reply:
Since there is currently no other hint of a distinction between `StatefulBrowser` and `Browser` methods I decided to omit the whole sentence.